lib/graph/function/ints_comparison.rb

```diff
@@ -1,48 +1,13 @@
 module Graph
   module Function
-    class IntsComparison
-      extend ReformatString
-      extend PlotConfig
-
-      def self.of(method_one, method_two)
-        fail unless method_one.is_a?(Method) && method_two.is_a?(Method)
-
-        Gnuplot.open do |gp|
-          Gnuplot::Plot.new(gp) do |plot|
-            self.class.send(:define_method, :a, proc(&method_one))
-            self.class.send(:define_method, :b, proc(&method_two))
-
-            plot.title  (title = "#{camel_title(method_one.name)} vs #{camel_title(method_two.name)}")
-            set_up(plot)
-
-            x = Graph::Function.configuration.step
-            pb = ProgressBar.create(title: title, total: x.size)
-
-            y = x.collect do |v|
-              pb.increment
-              array = (0..v - 1).to_a.shuffle
-              Benchmark.measure { a(array) }.real
-            end
-
-            plot.data << Gnuplot::DataSet.new( [x, y] ) do |ds|
-              ds.with = "linespoints"
-              ds.title = "#{escape_underscores(method_one.name)}"
-            end
-
-            pb.reset
-
-            z = x.collect do |v|
-              pb.increment
-              array = (0..v - 1).to_a.shuffle
-              Benchmark.measure { b(array) }.real
-            end
+    class IntsComparison < Comparison
+      def initialize
+        @data_generator = Proc.new {|v| (0..v - 1).to_a.shuffle }
```

Comment:
To be the same size it'd have to be like (-v/2 .. v/2) but I think there's not necessarily a need here -- if negatives are needed one could just write a generator.